### PR TITLE
fix(cli): reconcile history export not-found semantics, extract shared loader

### LINKED ISSUE #7156: Reconcile empty-conversation handling in readCliHistoryExportInput with history show's not-found semantics
## Context

Follow-up from #7059 (feat(cli): add `--export html|md` to `history show`).

`readCliHistoryExportInput` (`packages/cli/src/runtime/history.ts:240-242`) treats a stored empty `conversation` array (`[]`) as present, because `![]` is `false`:

```ts
if (!meta && !config && !conversation) return { status: 'not_found' };
if (!config || !conversation) return { status: 'incomplete' };
```

So `(config: null, conversation: [], meta: null)` reports `'incomplete'` ("execution exists, nothing to export yet"). But `readCliHistoryDetails` — the function backing plain `history show` — builds no preview from an empty conversation array, and with `config`/`meta` also null all of its guard fields are falsy, so it reports the *same* execution id as not found.

This means `history show <id>` and `history show <id> --export html` can disagree about whether an execution exists, for the same id and `--cwd`. Flagged by Cursor Bugbot during review; left open as non-blocking/cosmetic since the state is hard to reach in practice (config is normally written before conversation).

## Details

- Compare `readCliHistoryExportInput`'s not-found/incomplete branching against `readCliHistoryDetails`'s notion of "not found" and align them — likely by treating an empty `conversation` array the same as absent when deciding `not_found` vs `incomplete`.
- Add a regression test covering `(config: null, conversation: [], meta: null)` (and any other combination where the two functions currently disagree).

## References

- PR: #7059
- Parent: #7155
- Review comment (Cursor Bugbot): `packages/cli/src/runtime/history.ts#L240-L242`

### LINKED ISSUE #7157: Extract a shared ChatExportInput loader for the CLI and extension chat-export paths
## Context

Follow-up from #7059 (feat(cli): add `--export html|md` to `history show`).

`readCliHistoryExportInput` (`packages/cli/src/runtime/history.ts:232-261`) reads `config`/`conversation`/`meta` from the execution store and builds a `ChatExportInput` — deliberately mirroring `ChatExportController.buildExportInput` (`src/controllers/settingsView/ChatExportController.ts:89-123`) so the CLI and extension render conversations identically. That mirroring now means the same read/validate/assemble logic is duplicated verbatim across two call sites. Raised as a non-blocking suggestion across multiple review rounds on #7059 (worth extracting "at some point," per CLAUDE.md's factory-pattern guidance: justified once a duplicate exists in more than one place with real logic, not just spread).

## Details

- Extract a shared, host-neutral helper (e.g. `loadChatExportInput`) that both `readCliHistoryExportInput` and `ChatExportController.buildExportInput` call, so the read/validate/assemble logic has one source of truth.
- Keep the helper in a VS Code-free zone (e.g. alongside `src/agent/export/`) since the CLI runtime must not depend on `vscode`.
- Update both call sites and their existing test coverage to go through the shared helper.

## References

- PR: #7059
- Parent: #7155

### PR DESCRIPTION
**Pull request by LionSR**
Closes #7156, closes #7157

## What changed

**#7156** — `readCliHistoryExportInput` treated a stored empty conversation array (`[]`) as *present*, because `![]` is `false` in JS:

```ts
if (!meta && !config && !conversation) return { status: 'not_found' };
if (!config || !conversation) return { status: 'incomplete' };
```

So `(config: null, conversation: [], meta: null)` reported `'incomplete'` ("exists, nothing to export yet"). But `readCliHistoryDetails` (backing plain `history show`) builds no preview from an empty array, and with config/meta also null, reports the *same* execution id as not found. `history show <id>` and `history show <id> --export html` disagreed about existence for the same id.

**#7157** — `readCliHistoryExportInput` deliberately mirrored `ChatExportController.buildExportInput`: both read config/conversation/meta from the execution store and build a `ChatExportInput`, so the CLI and extension render conversations identically. That mirroring was duplicated read/validate/assemble logic across two call sites.

## Fix

Extracted a shared, host-neutral `loadChatExportInput` helper (`src/agent/export/loadChatExportInput.ts`) that does the store reads, normalizes a stored-but-empty conversation array to `null` (matching "no conversation" semantically instead of by truthiness), and assembles the `ChatExportInput` when both config and a non-empty conversation are present.

Both callers are now thin, host-specific wrappers:
- `readCliHistoryExportInput` (CLI) derives `not_found` / `incomplete` / `ok` from the loader's facts.
- `ChatExportController.buildExportInput` (extension) derives `config_missing` / `conversation_missing` / `ok`, and also drops a redundant `AgentConfigSchema.parse()` re-parse of a value that `store.readConfig()` already validates.

Both hosts get the corrected empty-array semantics for free, and `history show <id>` / `history show <id> --export html` now agree on existence for the same id.

## Tests

- `src/test-kernel/cli/History.vitest.mts` — added coverage for the empty-array reconciliation (`not_found` when nothing else exists, `incomplete` when config exists) and confirmed `readCliHistoryDetails` agrees.
- `src/test-kernel/agent/export/loadChatExportInput.vitest.ts` — new, direct coverage of the shared loader (assemble, nothing stored, empty-array normalization, config-only, conversation-only).
- `src/test-kernel/controllers/ChatExportController.vitest.ts` — new, covers the extension wrapper's status derivation including the empty-array case.

## Verification

- `npm run typecheck` — clean
- Targeted vitest files (`History.vitest.mts`, `loadChatExportInput.vitest.ts`, `ChatExportController.vitest.ts`, plus pre-existing `chatExportFormatter.vitest.ts` / `HistoryHandlers.vitest.ts` as a smoke check) — all pass
- `eslint` on touched files — clean (no errors, no warnings)

Co-Authored-By: Claude Sonnet 5 <noreply@anthropic.com>

---

> [!NOTE]
> **Low Risk**
> Refactor plus edge-case semantics for export/history only; no auth, payments, or core agent runtime changes.
> 
> **Overview**
> Introduces shared **`loadChatExportInput`** so CLI history export and the extension Settings chat export read the same execution-store data and build identical **`ChatExportInput`** payloads. **`readCliHistoryExportInput`** and **`ChatExportController.buildExportInput`** are now thin wrappers that map loader results to host-specific statuses; the extension no longer re-parses config with **`AgentConfigSchema`**.
> 
> **Empty conversation fix:** stored **`[]`** is normalized to “no conversation” (not truthy-present), so **`history show`** and **`history show --export`** agree on **not_found** vs **incomplete** for the same execution id. Tests cover the shared loader, CLI export semantics, and the extension wrapper.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 654ed491f55454175d356714e5f85d3bf6ef86a2. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/agent/export/loadChatExportInput.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+ * own status vocabulary (see `readCliHistoryExportInput` in
+ * `packages/cli/src/runtime/history.ts` and
+ * `ChatExportController.buildExportInput` in
+ * `src/controllers/settingsView/ChatExportController.ts`).
+ *
+ * `store.readConfig()` already validates against `AgentConfigSchema`
+ * internally and falls back to `null` on a schema mismatch (see
+ * `ExecutionKVStore.readValidated`), so `config` here is never a raw,
+ * unvalidated value — no redundant re-parse (and no risk of it throwing on a
+ * corrupt record) is needed.
```
[the remark this answers is not in the document]
Comment (LionSR) [reply src/agent/export/loadChatExportInput.ts]: Fixed: reworded to point at the private readValidated helper on StorageFSKVStore (not the ExecutionKVStore interface). See 654ed491f.

File: src/agent/export/loadChatExportInput.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+/**
+ * A stored conversation is only "present" when it has at least one message.
+ * A stored-but-empty array must be treated the same as absent so that
+ * every existence check downstream (this module's `exportInput`, and each
+ * host's own not-found/incomplete classification) agrees with
+ * `readCliHistoryDetails`, which builds no preview from an empty array
+ * either. Without this, a plain truthiness check (`!conversation`) would
+ * treat `[]` as "present" — `![]` is `false` in JS — and misreport an
+ * execution `history show` already treats as not found.
+ */
```
[the remark this answers is not in the document]
Comment (LionSR) [reply src/agent/export/loadChatExportInput.ts:55]: Fixed: reworded to make clear this is a defensive/contract-enforcing check (ExecutionKVStore.readConversation already normalizes empty arrays to null at the store layer) rather than the primary fix at this call site, and noted the tests-mock-the-store-directly rationale for keeping it. See 654ed491f.
